Seems like this isn't quite working yet?

```
$ npm i

> srand@1.1.4 install /Users/isaacs/dev/js/srand
> node-gyp rebuild

  CXX(target) Release/obj.target/srand/srand.o
../srand.cc:14:3: error: use of undeclared identifier 'NanScope'
  NanScope();
  ^
../srand.cc:15:18: error: use of undeclared identifier 'NanNew'
  NanReturnValue(NanNew<Number>(rand()/((double)RAND_MAX + 1)));
                 ^
../srand.cc:15:25: error: 'Number' does not refer to a value
  NanReturnValue(NanNew<Number>(rand()/((double)RAND_MAX + 1)));
                        ^
/Users/isaacs/.node-gyp/5.6.0/deps/v8/include/v8.h:2480:17: note: declared here
class V8_EXPORT Number : public Primitive {
                ^
../srand.cc:19:3: error: use of undeclared identifier 'NanScope'
  NanScope();
  ^
../srand.cc:20:18: error: use of undeclared identifier 'NanNew'
  NanReturnValue(NanNew<Integer>(rand()));
                 ^
../srand.cc:20:25: error: 'Integer' does not refer to a value
  NanReturnValue(NanNew<Integer>(rand()));
                        ^
/Users/isaacs/.node-gyp/5.6.0/deps/v8/include/v8.h:2494:17: note: declared here
class V8_EXPORT Integer : public Number {
                ^
../srand.cc:24:3: error: use of undeclared identifier 'NanScope'
  NanScope();
  ^
../srand.cc:25:7: error: use of undeclared identifier 'args'
  if (args.Length() == 0) {
      ^
../srand.cc:26:36: error: use of undeclared identifier 'NanNew'
    NanThrowError(Exception::Error(NanNew<String>("Usage: srand(n)")));
                                   ^
../srand.cc:26:43: error: 'String' does not refer to a value
    NanThrowError(Exception::Error(NanNew<String>("Usage: srand(n)")));
                                          ^
/Users/isaacs/.node-gyp/5.6.0/deps/v8/include/v8.h:2078:17: note: declared here
class V8_EXPORT String : public Name {
                ^
../srand.cc:28:9: error: use of undeclared identifier 'args'
  srand(args[0]->Uint32Value());
        ^
../srand.cc:29:18: error: use of undeclared identifier 'NanUndefined'; did you mean 'Nan::Undefined'?
  NanReturnValue(NanUndefined());
                 ^~~~~~~~~~~~
                 Nan::Undefined
../node_modules/nan/nan.h:521:39: note: 'Nan::Undefined' declared here
  NAN_INLINE v8::Local<v8::Primitive> Undefined() {
                                      ^
../srand.cc:29:3: error: use of undeclared identifier 'NanReturnValue'
  NanReturnValue(NanUndefined());
  ^
../srand.cc:33:3: error: no matching function for call to 'NODE_SET_METHOD'
  NODE_SET_METHOD(target, "random", Random);
  ^~~~~~~~~~~~~~~
/Users/isaacs/.node-gyp/5.6.0/src/node.h:248:25: note: expanded from macro 'NODE_SET_METHOD'
#define NODE_SET_METHOD node::NODE_SET_METHOD
                        ^~~~~~~~~~~~~~~~~~~~~
/Users/isaacs/.node-gyp/5.6.0/src/node.h:236:13: note: candidate function [with TypeName = v8::Local<v8::Object>] not viable: no known conversion from
      'Nan::NAN_METHOD_RETURN_TYPE (Nan::NAN_METHOD_ARGS_TYPE)' to 'v8::FunctionCallback' (aka 'void (*)(const FunctionCallbackInfo<v8::Value> &)') for 3rd
      argument
inline void NODE_SET_METHOD(const TypeName& recv,
            ^
../srand.cc:34:3: error: no matching function for call to 'NODE_SET_METHOD'
  NODE_SET_METHOD(target, "rand", Rand);
  ^~~~~~~~~~~~~~~
/Users/isaacs/.node-gyp/5.6.0/src/node.h:248:25: note: expanded from macro 'NODE_SET_METHOD'
#define NODE_SET_METHOD node::NODE_SET_METHOD
                        ^~~~~~~~~~~~~~~~~~~~~
/Users/isaacs/.node-gyp/5.6.0/src/node.h:236:13: note: candidate function [with TypeName = v8::Local<v8::Object>] not viable: no known conversion from
      'Nan::NAN_METHOD_RETURN_TYPE (Nan::NAN_METHOD_ARGS_TYPE)' to 'v8::FunctionCallback' (aka 'void (*)(const FunctionCallbackInfo<v8::Value> &)') for 3rd
      argument
inline void NODE_SET_METHOD(const TypeName& recv,
            ^
../srand.cc:35:3: error: no matching function for call to 'NODE_SET_METHOD'
  NODE_SET_METHOD(target, "srand", SRand);
  ^~~~~~~~~~~~~~~
/Users/isaacs/.node-gyp/5.6.0/src/node.h:248:25: note: expanded from macro 'NODE_SET_METHOD'
#define NODE_SET_METHOD node::NODE_SET_METHOD
                        ^~~~~~~~~~~~~~~~~~~~~
/Users/isaacs/.node-gyp/5.6.0/src/node.h:236:13: note: candidate function [with TypeName = v8::Local<v8::Object>] not viable: no known conversion from
      'Nan::NAN_METHOD_RETURN_TYPE (Nan::NAN_METHOD_ARGS_TYPE)' to 'v8::FunctionCallback' (aka 'void (*)(const FunctionCallbackInfo<v8::Value> &)') for 3rd
      argument
inline void NODE_SET_METHOD(const TypeName& recv,
            ^
../srand.cc:36:3: error: no matching function for call to 'NODE_SET_METHOD'
  NODE_SET_METHOD(target, "seed", SRand);
  ^~~~~~~~~~~~~~~
/Users/isaacs/.node-gyp/5.6.0/src/node.h:248:25: note: expanded from macro 'NODE_SET_METHOD'
#define NODE_SET_METHOD node::NODE_SET_METHOD
                        ^~~~~~~~~~~~~~~~~~~~~
/Users/isaacs/.node-gyp/5.6.0/src/node.h:236:13: note: candidate function [with TypeName = v8::Local<v8::Object>] not viable: no known conversion from
      'Nan::NAN_METHOD_RETURN_TYPE (Nan::NAN_METHOD_ARGS_TYPE)' to 'v8::FunctionCallback' (aka 'void (*)(const FunctionCallbackInfo<v8::Value> &)') for 3rd
      argument
inline void NODE_SET_METHOD(const TypeName& recv,
            ^
17 errors generated.
make: *** [Release/obj.target/srand/srand.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/isaacs/dev/npm/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:100:13)
gyp ERR! stack     at ChildProcess.emit (events.js:185:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.3.0
gyp ERR! command "/usr/local/bin/node" "/Users/isaacs/dev/npm/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/isaacs/dev/js/srand
gyp ERR! node -v v5.6.0
gyp ERR! node-gyp -v v3.2.1
gyp ERR! not ok

npm ERR! Darwin 15.3.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "i"
npm ERR! node v5.6.0
npm ERR! npm  v3.7.0
npm ERR! code ELIFECYCLE
npm ERR! srand@1.1.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the srand@1.1.4 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the srand package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs srand
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls srand
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,5 @@
 .lock-wscript
 build/
+
+node_modules/
+.node-version

--- a/package.json
+++ b/package.json
@@ -9,6 +9,6 @@
   },
   "license": "ISC",
   "dependencies": {
-    "nan": "^1.7.0"
+    "nan": "^2.2.0"
   }
 }

--- a/srand.cc
+++ b/srand.cc
@@ -1,40 +1,32 @@
-
 #include <stdlib.h>
-#include <v8.h>
 #include <nan.h>
 #include <node.h>
 #include <cstdlib>
 
-using namespace node;
-using namespace v8;
-
-extern "C" void init (Handle<Object>);
-
-NAN_METHOD(Random) {
-  NanScope();
-  NanReturnValue(NanNew<Number>(rand()/((double)RAND_MAX + 1)));
+void Random(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Local<v8::Number> num = Nan::New(rand()/((double)RAND_MAX + 1));
+  info.GetReturnValue().Set(num);
 }
 
-NAN_METHOD(Rand) {
-  NanScope();
-  NanReturnValue(NanNew<Integer>(rand()));
+void Rand(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Local<v8::Integer> num = Nan::New(rand());
+  info.GetReturnValue().Set(num);
 }
 
-NAN_METHOD(SRand) {
-  NanScope();
-  if (args.Length() == 0) {
-    NanThrowError(Exception::Error(NanNew<String>("Usage: srand(n)")));
+void SRand(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  if (info.Length() == 0) {
+    Nan::ThrowError("Usage: srand(n)");
   }
-  srand(args[0]->Uint32Value());
-  NanReturnValue(NanUndefined());
+  srand(info[0]->Uint32Value());
+  info.GetReturnValue().SetUndefined();
 }
 
-extern "C" void init (Handle<Object> target) {
-  NODE_SET_METHOD(target, "random", Random);
-  NODE_SET_METHOD(target, "rand", Rand);
-  NODE_SET_METHOD(target, "srand", SRand);
-  NODE_SET_METHOD(target, "seed", SRand);
-  NODE_DEFINE_CONSTANT(target, RAND_MAX);
+void Init(v8::Local<v8::Object> exports) {
+  exports->Set(Nan::New("random").ToLocalChecked(), Nan::New<v8::FunctionTemplate>(Random)->GetFunction());
+  exports->Set(Nan::New("rand").ToLocalChecked(), Nan::New<v8::FunctionTemplate>(Rand)->GetFunction());
+  exports->Set(Nan::New("srand").ToLocalChecked(), Nan::New<v8::FunctionTemplate>(SRand)->GetFunction());
+  exports->Set(Nan::New("seed").ToLocalChecked(), Nan::New<v8::FunctionTemplate>(SRand)->GetFunction());
+  exports->Set(Nan::New("RAND_MAX").ToLocalChecked(), Nan::New<v8::Integer>(RAND_MAX));
 }
 
-NODE_MODULE(srand, init)
+NODE_MODULE(srand, Init)

